source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -0,0 +1,100 @@
+#include <fcntl.h>
+#include <io.h>
+#include <sys/stat.h>
+#include <windows.h>
+
+#undef DELETE
+#undef GetMessage
+
+#include "common/common/assert.h"
+#include "common/filesystem/filesystem_impl.h"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "common/common/fmt.h"
+
+namespace Envoy {
+namespace Filesystem {
+
+FileImplWin32::~FileImplWin32() {
+  if (isOpen()) {
+    const Api::IoCallBoolResult result = close();
+    ASSERT(result.rc_);
+  }
+}
+
+void FileImplWin32::openFile() {
+  const int flags = _O_RDWR | _O_APPEND | _O_CREAT;
+  const int mode = _S_IREAD | _S_IWRITE;
+
+  fd_ = ::_open(path_.c_str(), flags, mode);
+}
+
+ssize_t FileImplWin32::writeFile(absl::string_view buffer) {
+  if (fd_ == -1) {
+    errno = EBADF;
```

Comment:
If I understand the approach correctly here this is setting the "real" `errno` to -1. Is this to replicate the POSIX behavior?

Reply:
`EBADF` is actually defined on Windows. This is to replicate the POSIX behavior.